ironfish/src/wallet/database/accountsdb.ts

```diff
@@ -181,9 +181,10 @@ export class AccountsDB {
     })
   }
 
-  async removeAccount(id: string, tx?: IDatabaseTransaction): Promise<void> {
+  async removeAccount(account: Account, tx?: IDatabaseTransaction): Promise<void> {
     await this.database.withTransaction(tx, async (tx) => {
-      await this.accounts.del(id, tx)
+      await this.accounts.del(account.id, tx)
+      await this.balances.del(account.id, tx)
```

Comment:
Should we clear other account data here?

Reply:
Maybe, but that'll also take a looong time and freeze the node on a big wallet. We need to implement time sliced removal where we mark the account as deleted, then have something in the background that cleans it up over time until its all gone.